mlrun/model_monitoring/db/tsdb/v3io/v3io.py

```diff
@@ -0,0 +1,365 @@
+        self.tables = {}
+
+        events_table_full_path = mlrun.mlconf.get_model_monitoring_file_target_path(
+            project=self.project,
+            kind=mm_constants.FileTargetKind.EVENTS,
+        )
+        (
+            _,
+            _,
+            events_path,
+        ) = mlrun.common.model_monitoring.helpers.parse_model_endpoint_store_prefix(
+            events_table_full_path
+        )
+        self.tables[mm_constants.V3IOTSDBTables.EVENTS] = events_path
+
+        monitoring_application_full_path = (
+            mlrun.mlconf.get_model_monitoring_file_target_path(
+                project=self.project,
+                kind=mm_constants.FileTargetKind.MONITORING_APPLICATION,
+            )
+        )
+        (
+            _,
+            _,
+            monitoring_application_path,
+        ) = mlrun.common.model_monitoring.helpers.parse_model_endpoint_store_prefix(
+            monitoring_application_full_path
+        )
+        self.tables[mm_constants.V3IOTSDBTables.APP_RESULTS] = (
+            monitoring_application_path + mm_constants.V3IOTSDBTables.APP_RESULTS
+        )
+        self.tables[mm_constants.V3IOTSDBTables.METRICS] = (
+            monitoring_application_path + mm_constants.V3IOTSDBTables.METRICS
+        )
+
+    def create_tsdb_application_tables(self):
+        """
+        Create the application tables using the TSDB connector. At the moment we support 2 types of application tables:
+        - app_results: a detailed result that includes status, kind, extra data, etc.
+        - metrics: a basic key value that represents a single numeric metric.
+        """
+        application_tables = [
+            mm_constants.V3IOTSDBTables.APP_RESULTS,
+            mm_constants.V3IOTSDBTables.METRICS,
+        ]
+        for table in application_tables:
+            logger.info("Creating table in V3IO TSDB", table=table)
+            self._frames_client.create(
+                backend=_TSDB_BE,
+                table=self.tables[table],
+                if_exists=IGNORE,
+                rate=_TSDB_RATE,
+            )
+
+    def apply_monitoring_stream_steps(
+        self,
+        graph,
+        tsdb_batching_max_events: int = 10,
+        tsdb_batching_timeout_secs: int = 300,
+    ):
+        """
+        Apply TSDB steps on the provided monitoring graph. Throughout these steps, the graph stores live data of
+        different key metric dictionaries.This data is being used by the monitoring dashboards in
+        grafana. Results can be found under  v3io:///users/pipelines/project-name/model-endpoints/events/.
+        In that case, we generate 3 different key  metric dictionaries:
+        - base_metrics (average latency and predictions over time)
+        - endpoint_features (Prediction and feature names and values)
+        - custom_metrics (user-defined metrics
+        """
+
+        # Step 12 - Before writing data to TSDB, create dictionary of 2-3 dictionaries that contains
```

Comment:
It doesn't seem like this Step X stuff belong in the connector class, and could easily become wrong if someone adds a step earlier in the flow, elsewhere in the code.

Reply:
Got it, I'll update the description and remove the "hard-coded" step numbering.